lib/cisco_node_utils/itd_device_group.rb

```diff
@@ -0,0 +1,210 @@
+# February 2016, Sai Chintalapudi
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for itd_device_group
+  class ItdDeviceGroup < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name.length > 0
+      @name = name
+
+      set_args_keys_default
+      create if instantiate
+    end
+
+    def self.itds
+      hash = {}
+      group_list = []
+      groups = config_get('itd_device_group',
+                          'all_itd_device_groups')
+      return hash if groups.nil?
+
+      groups.each do |group|
+        # The show cmd shows more than name if the probe is not confgured yet
+        # itd device-group abc frequency 10 timeout 5 retry-down-count 3 ...
+        # so filter it out to just get the name
+        group_list << group.split[0]
+      end
+
+      group_list.each do |id|
+        hash[id] = ItdDeviceGroup.new(id, false)
+      end
+      hash
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def create
+      Feature.itd_enable
+      config_set('itd_device_group', 'create', name: @name)
+    end
+
+    def destroy
+      config_set('itd_device_group', 'destroy', name: @name)
+    end
+
+    # Helper method to delete @set_args hash keys
+    def set_args_keys_default
+      @set_args = { name: @name }
+      @get_args = @set_args
+    end
+
+    def probe_params
+      params = config_get('itd_device_group', 'probe', @get_args)
```

Comment:
This should be simplified to:
`config_get('itd_device_group', 'probe', @get_args)`